test/manifolds/test_sphere.jl

```diff
@@ -0,0 +1,356 @@
+s = joinpath(@__DIR__, "..", "ManifoldsTestSuite.jl")
+!(s in LOAD_PATH) && (push!(LOAD_PATH, s))
+using ManifoldsTestSuite
+
+using LinearAlgebra, Manifolds, Test, StaticArrays, Quaternions, Random, Distributions, RecursiveArrayTools
+using ManifoldDiff
+
+
+@testset "The Spheres" begin
+
+    M = Sphere(2)
+    p = [1.0, 0.0, 0.0]
+    q = [1 / sqrt(2), 1 / sqrt(2), 0.0]
+    X = [0.0, π / 4, 0.0]
+    Y = [0.0, 0.0, π / 4]
+    V = [2.0, 0.0, 0.0]  # normal vector at p
+    ξ = [0.0, π / 4, 0.0]  # covector at p
+
+    @testset "Basics" begin
+        @test is_default_metric(M, EuclideanMetric())
+        @test !is_default_metric(M, AffineInvariantMetric())
+        @test base_manifold(M) === M
+        @test representation_size(M) == (3,)
+        @test !is_flat(M)
+        @test is_flat(Sphere(1))
```

Comment:
We should discuss which of these do make sense to be moved to the test suite.

Reply:
`representation_size` and `is_flat` definitely make sense to be in the suite, I'm not sure about the other ones.